compiler/noirc_frontend/src/hir/comptime/interpreter.rs

```diff
@@ -609,10 +609,12 @@ impl<'a> Interpreter<'a> {
         let rhs = self.evaluate(infix.rhs)?;
 
         // TODO: Need to account for operator overloading
-        assert!(
-            self.interner.get_selected_impl_for_expression(id).is_none(),
-            "Operator overloading is unimplemented in the interpreter"
-        );
+        if self.interner.get_selected_impl_for_expression(id).is_some() {
+            return Err(InterpreterError::Unimplemented {
+                item: "Operator overloading is unimplemented in the interpreter. See https://github.com/noir-lang/noir/issues/4925".to_string(),
```

Comment:
```suggestion
                item: "Operator overloading in the interpreter".to_string(),
```
The Unimplemented error will display the error as `format!("{item} is currently unimplemented")`

Reply:
I didn't see a good spot to add the issue number - I suppose we could add it as an optional field to the error variant.